src/nupic/types/Sdr.cpp

```diff
@@ -360,6 +360,62 @@ namespace sdr {
         SDR::setDenseInplace();
     }
 
+    void SparseDistributedRepresentation::concatenate(const SDR &inp1, const SDR &inp2, UInt axis)
+        { concatenate({&inp1, &inp2}, axis); }
+
+    void SparseDistributedRepresentation::concatenate(vector<const SDR*> inputs, UInt axis)
+    {
+        // Check inputs.
+        NTA_CHECK( inputs.size() >= 2u )
+            << "Not enough inputs to SDR::concatenate, need at least 2 SDRs got " << inputs.size() << "!";
+        UInt axis_dim_sum = 0u;
+        for( const auto &sdr : inputs ) {
+            NTA_CHECK( sdr != nullptr );
+            NTA_CHECK( sdr->dimensions.size() == dimensions.size() )
+                << "All inputs to SDR::concatenate must have the same number of dimensions as the output SDR!";
+            for( auto dim = 0u; dim < dimensions.size(); dim++ ) {
+                if( dim == axis ) {
+                    axis_dim_sum += sdr->dimensions[dim];
```

Comment:
- suggestion: call it `concat_size` instead of axis_dim_sum? 
- nit but more readable: `+= sdr->dimensions[axis];`